### PR TITLE
fix the nested target map

### PR DESCRIPTION
**Pull request by sallyom**
@cooktheryan @josephsawaya as noted, there were some flaws in the initial refactoring.
* This PR hopefully fixes those flaws
* finishes the renaming of repo/Repo/Repos -> target/Target/Targets
* finishes the flattening of Repos.Targets to Targets (that nested map thing is gone)
* fixes FindDiff, since it was broken

### CODE REVIEW

File: pkg/engine/harpoon.go [hunk 1]
```diff
@@ -151,26 +151,24 @@ func (hc *HarpoonConfig) getTargets() {
...
-	// single map.
+
 	s := gocron.NewScheduler(time.UTC)
-	// TODO: can tag jobs, use tags to track
-	for methodRepo, schedule := range allTargets {
-		methodAndRepo := strings.Split(methodRepo, split)
-		method := methodAndRepo[0]
-		repoName := methodAndRepo[1]
+	// method, schedule := range hc.RepoTargetMap[repo.Name]
+	// repoName, map[method]schedule := range allTargets
```
Comment (sallyom) [inline pkg/engine/harpoon.go:165]: comments so I don't have to strain to wrap my brain around this in the future.

File: pkg/engine/api/types.go [hunk 2]
```diff
@@ -14,31 +14,33 @@ type Target struct {
 	Systemd      Systemd      `mapstructure:"systemd"`
 	Kube         Kube         `mapstructure:"kube"`
 	FileTransfer FileTransfer `mapstructure:"fileTransfer"`
-	Mu           sync.Mutex
+	// map of method:schedule
```
Comment (sallyom) [inline pkg/engine/api/types.go:17]: we can remove this note in a follow-up, or if I repush anything here

File: pkg/engine/harpoon.go [hunk 3]
```diff
@@ -234,57 +228,60 @@ func (hc *HarpoonConfig) processRaw(ctx context.Context, repo *api.Target, sched
...
 				}
 				return nil
 			})
 		}
 	}
-	changes, isDiff := hc.findDiff(gitRepo, directory, rawMethod, repo.Branch, initial)
-	if !isDiff && !initial {
-		klog.Infof("Repo: %s, Method: %s: Nothing to pull.....Requeuing", repo.Name, rawMethod)
+	changes := hc.findDiff(gitRepo, directory, rawMethod, target.Branch)
+	// if first run, don't want to return here
```
Comment (sallyom) [inline pkg/engine/harpoon.go:249]: this note is outdated, need to remove it - I'll let the initial test run finish to be sure things are working, then will push a fix to remove the comment

**Comment by sallyom** [comment]
nice, all tests (except change) passed - I'll confirm the change validate passes locally

File: pkg/engine/api/types.go (hunk 2 above)
Comment (sallyom) [reply pkg/engine/api/types.go]: removed it

File: pkg/engine/harpoon.go (hunk 3 above)
Comment (sallyom) [reply pkg/engine/harpoon.go]: removed it

**Review by cooktheryan — APPROVED** [review]

**Pull request merged**